WordPress/Classes/ViewRelated/Post/PageMenuViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+
+final class PageMenuViewModel: AbstractPostMenuViewModel {
+
+    private let page: Page
+    private let isJetpackFeaturesEnabled: Bool
+    private let isBlazeFlagEnabled: Bool
+
+    var buttonSections: [AbstractPostButtonSection] {
+        [
+            createPrimarySection(),
+            createSecondarySection(),
+            createBlazeSection(),
+            createTrashSection()
+        ]
+    }
+
+    init(
+        page: Page,
+        isJetpackFeaturesEnabled: Bool = JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled(),
+        isBlazeFlagEnabled: Bool = BlazeHelper.isBlazeFlagEnabled()
+    ) {
+        self.page = page
+        self.isJetpackFeaturesEnabled = isJetpackFeaturesEnabled
+        self.isBlazeFlagEnabled = isBlazeFlagEnabled
+    }
+
+    private func createPrimarySection() -> AbstractPostButtonSection {
+        var buttons = [AbstractPostButton]()
+
+        if !page.isFailed {
+            buttons.append(.view)
+        }
+
+        return AbstractPostButtonSection(buttons: buttons)
+    }
+
+    private func createSecondarySection() -> AbstractPostButtonSection {
+        var buttons = [AbstractPostButton]()
+
+        if page.status != .draft {
+            buttons.append(.moveToDraft)
```

Comment:
(Unrelated) wp.com uses the wording "Restore" when the post is trashed. Maybe something the app could also adopt?